File: examples/javascript/chat_non_stream.js
"use strict";

/**
 * Node 18+ example.
 * Browser usage is similar with fetch(), but you must configure CORS on server.
 */

const baseUrl = (process.env.NESTY_BASE_URL || "http://127.0.0.1:8000").replace(/\/+$/, "");
const apiKey = (process.env.NESTY_API_KEY || "").trim();
const model = (process.env.NESTY_MODEL || "nesty-combined-1.0").trim();

async function main() {
  const headers = { "Content-Type": "application/json" };
  if (apiKey) headers.Authorization = `Bearer ${apiKey}`;

  // Parse optional environment configurations
  const storeVal = (process.env.NESTY_STORE || "false").trim().toLowerCase() === "true";
  const searchVal = (process.env.NESTY_SEARCH || "off").trim().toLowerCase();
  const toolsVal = (process.env.NESTY_TOOLS || "off").trim().toLowerCase();
  const semanticRecallVal = (process.env.NESTY_SEMANTIC_RECALL || "auto").trim().toLowerCase();

  const payload = {
    model,
    messages: [{ role: "user", content: "Write a short intro about NestyAI." }],
    stream: false,
    store: storeVal,
    search: searchVal,
    tools: toolsVal,
    semantic_recall: semanticRecallVal,
  };

  let response;
  try {
    response = await fetch(`${baseUrl}/v1/chat/completions`, {
      method: "POST",
      headers,
      body: JSON.stringify(payload),
    });
  } catch (err) {
    console.error("[ERROR] request failed:", err);
    process.exitCode = 1;
    return;
  }

  if (!response.ok) {
    const text = await response.text();
    console.error(`[ERROR] status=${response.status}`);
    console.error(text);
    process.exitCode = 1;
    return;
  }

  const data = await response.json();
  const content = data?.choices?.[0]?.message?.content;
  if (typeof content !== "string") {
    console.error("[ERROR] unexpected response shape");
    console.error(JSON.stringify(data, null, 2));
    process.exitCode = 1;
    return;
  }

  console.log(content);
}

main();

File: examples/python/chat_non_stream.py
from __future__ import annotations

import json
import os
import sys

import httpx


def main() -> int:
    base_url = os.getenv("NESTY_BASE_URL", "http://127.0.0.1:8000").rstrip("/")
    api_key = os.getenv("NESTY_API_KEY", "").strip()
    model = os.getenv("NESTY_MODEL", "nesty-combined-1.0").strip()

    headers = {"Content-Type": "application/json"}
    if api_key:
        headers["Authorization"] = f"Bearer {api_key}"

    # Parse optional environment configurations
    store_val = os.getenv("NESTY_STORE", "false").strip().lower() in {"1", "true", "yes", "on"}
    search_val = os.getenv("NESTY_SEARCH", "off").strip().lower()
    tools_val = os.getenv("NESTY_TOOLS", "off").strip().lower()
    semantic_recall_val = os.getenv("NESTY_SEMANTIC_RECALL", "auto").strip().lower()

    payload = {
        "model": model,
        "messages": [{"role": "user", "content": "Write a short intro about NestyAI."}],
        "stream": False,
        "store": store_val,
        "search": search_val,
        "tools": tools_val,
        "semantic_recall": semantic_recall_val,
    }

    try:
        with httpx.Client(timeout=30.0) as client:
            response = client.post(f"{base_url}/v1/chat/completions", json=payload, headers=headers)
    except Exception as exc:
        print(f"[ERROR] request failed: {exc}")
        return 1

    if response.status_code != 200:
        try:
            body = response.json()
        except Exception:
            body = {"raw": response.text}
        print(f"[ERROR] status={response.status_code}")
        print(json.dumps(body, ensure_ascii=False, indent=2))
        return 1

    data = response.json()
    try:
        content = data["choices"][0]["message"]["content"]
    except Exception:
        print("[ERROR] unexpected response format")
        print(json.dumps(data, ensure_ascii=False, indent=2))
        return 1

    print(content)
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
